cfsdk/src/main/java/io/harness/cfsdk/CfClient.java

```diff
@@ -298,88 +298,95 @@ public boolean unregisterEventsListener(final EventsListener observer) {
 
     public boolean boolVariation(String evaluationId, boolean defaultValue) {
 
-        final Evaluation evaluation = getEvaluationById(
+        final Evaluation evaluation = getEvaluationById(evaluationId, target);
 
-                evaluationId,
-                target,
-                defaultValue
-        );
-
-        final Object value = evaluation.getValue();
-        if (value instanceof Boolean) {
-
-            return (Boolean) value;
+        if (evaluation == null) {
+            log.warn("Evaluation for {} is null, returning default value", evaluationId);
+            SdkCodes.warnDefaultVariationServed(evaluationId, target, String.valueOf(defaultValue));
+            return defaultValue;
         }
-        if (value instanceof String) {
```

Comment:
All of these `instanceof` checks were scattered throughout these methods. They are not needed, as `getEvaluationById` always returns a string.